src/V3Unknown.cpp

```diff
@@ -140,6 +141,30 @@ class UnknownVisitor final : public VNVisitor {
         }
     }
 
+    AstVar* createAddTemp(const AstNodeExpr* const nodep) {
+        AstVar* const varp = new AstVar{nodep->fileline(), VVarType::XTEMP,
+                                        m_xrandNames->get(nodep), nodep->dtypep()};
+        if (m_ftaskp) {
+            varp->funcLocal(true);
+            varp->lifetime(VLifetime::AUTOMATIC_EXPLICIT);
+            m_ftaskp->stmtsp()->addHereThisAsNext(varp);
+        } else {
+            m_modp->stmtsp()->addHereThisAsNext(varp);
+        }
+        return varp;
+    }
+
+    // Returns true if it is known at compile time that `msbConstp` is greater than or equal
+    // `exprp`
+    static bool isStaticlyGte(AstConst* const msbConstp, const AstNodeExpr* const exprp) {
+        bool staticGte = msbConstp->width() >= exprp->width()
+                         && msbConstp->num().toSInt() >= (1 << exprp->width()) - 1;
```

Comment:
can `if()` this and return true, and when happens avoid executing the code below (which isn't that slow but not instant).